servers/quarkus-secrets/src/main/java/org/projectnessie/server/catalog/secrets/SecretsProducers.java

```diff
@@ -15,21 +15,96 @@
  */
 package org.projectnessie.server.catalog.secrets;
 
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.runtime.StartupEvent;
 import io.smallrye.config.SmallRyeConfig;
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Instance;
 import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Singleton;
+import java.util.Locale;
+import org.projectnessie.catalog.secrets.ImmutableResolvingSecretsProvider;
 import org.projectnessie.catalog.secrets.ResolvingSecretsProvider;
+import org.projectnessie.catalog.secrets.SecretsManager;
 import org.projectnessie.catalog.secrets.SecretsProvider;
+import org.projectnessie.catalog.secrets.cache.CachingSecrets;
+import org.projectnessie.catalog.secrets.cache.CachingSecretsBackend;
+import org.projectnessie.catalog.secrets.cache.SecretsCacheConfig;
 import org.projectnessie.catalog.secrets.smallrye.SmallryeConfigSecretsManager;
+import org.projectnessie.quarkus.config.QuarkusSecretsCacheConfig;
+import org.projectnessie.quarkus.config.QuarkusSecretsConfig;
+import org.projectnessie.quarkus.providers.RepositoryId;
 
 public class SecretsProducers {
 
   @Produces
   @Singleton
-  public SecretsProvider secretsProvider(SmallRyeConfig smallRyeConfig) {
+  public SecretsProvider secretsProvider(
+      QuarkusSecretsConfig config,
+      SmallRyeConfig smallRyeConfig,
+      @Any Instance<SecretsManagerBuilder> secretsSupplierBuilders,
+      Instance<MeterRegistry> meterRegistry,
+      @RepositoryId String repositoryId) {
+    QuarkusSecretsConfig.SecretsSupplierType type = config.type();
+
+    SecretsProvider resolving =
+        buildResolvingSecretsProvider(smallRyeConfig, secretsSupplierBuilders, type);
+
+    return maybeCache(config, meterRegistry, repositoryId, resolving);
+  }
+
+  private SecretsProvider buildResolvingSecretsProvider(
+      SmallRyeConfig smallRyeConfig,
+      Instance<SecretsManagerBuilder> secretsSupplierBuilders,
+      QuarkusSecretsConfig.SecretsSupplierType type) {
+
     // Reference secrets via `urn:nessie-secret:quarkus:<secret-name>
-    return ResolvingSecretsProvider.builder()
-        .putSecretsManager("quarkus", new SmallryeConfigSecretsManager(smallRyeConfig))
-        .build();
+    ImmutableResolvingSecretsProvider.Builder providers =
+        ResolvingSecretsProvider.builder()
+            .putSecretsManager("quarkus", new SmallryeConfigSecretsManager(smallRyeConfig));
+
+    if (secretsSupplierBuilders.isUnsatisfied()) {
+      return providers.build();
+    }
+
+    Instance<SecretsManagerBuilder> selected =
+        secretsSupplierBuilders.select(new SecretsType.Literal(type));
+
+    if (selected.isUnsatisfied()) {
+      return providers.build();
```

Comment:
Why is this not an error? If `type` is set, but not backed by anything in runtime, it seems like a config mistake or deeper error to me.